test/extended/openstack/loadbalancers.go

```diff
@@ -382,6 +383,152 @@ var _ = g.Describe("[sig-installer][Suite:openshift/openstack][lb][Serial] The O
+				return []string{}
+			}
+			e2e.Logf("Found AllowedCIDRs: %v", lbListener.AllowedCIDRs)
+			return lbListener.AllowedCIDRs
+
+		}, "10s", "1s").Should(o.ConsistOf(allowAllAllowedCidrs), "Didn't find the expected allowed_cidrs '%v'", allowAllAllowedCidrs)
+		e2e.Logf("Found expected allowed_cidrs '%v' in Openstack LoadBalancer Listener '%s'", allowAllAllowedCidrs, listenerId)
+
+		// Wait until ProvisioningStatus in the LB listener is back to ACTIVE after it's been updated
+		activeProvisioningStatus := "ACTIVE"
+		e2e.Logf("Expected ProvisioningStatus: '%s'", activeProvisioningStatus)
+		o.Eventually(func() string {
+			lbListener, err := octavialisteners.Get(loadBalancerClient, listenerId).Extract()
+			if err != nil {
+				e2e.Logf("Error ocurred: %v, trying next iteration", err)
+				return ""
+			}
+			e2e.Logf("Found ProvisioningStatus: %s", lbListener.ProvisioningStatus)
+			return lbListener.ProvisioningStatus
+
+		}, "10s", "1s").Should(o.Equal(activeProvisioningStatus), "Didn't find the expected ProvisioningStatus '%s'", activeProvisioningStatus)
+		e2e.Logf("Found expected ProvisioningStatus '%s' in Openstack LoadBalancer Listener '%s'", activeProvisioningStatus, listenerId)
+
+		// Check there is still only one listener in the LB and that the Id matches with the previously stored one
+		allPages, err = octavialisteners.List(loadBalancerClient, octavialisteners.ListOpts{LoadbalancerID: loadBalancerId}).AllPages()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		listeners, err = octavialisteners.ExtractListeners(allPages)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(listeners).Should(o.HaveLen(1), "More than one listener found for LB '%s': %v", loadBalancerId, listeners)
+		o.Expect(listeners[0].ID).Should((o.Equal(listenerId)), "Listener ID for LB '%s' has changed", loadBalancerId)
+
+		// Check connectivity to the service
+		connNumber = 10
+		g.By(fmt.Sprintf("accessing the service %d times from outside and storing the name of the pods answering", connNumber))
+		results = make(map[string]int)
+		for i := 0; i < connNumber; i++ {
+			// https://github.com/kubernetes/kubernetes/blob/master/test/images/agnhost/README.md#netexec
+			podName, err := getPodNameThroughUdpLb(svcIp, fmt.Sprintf("%d", svcPort), "hostname")
+			if err != nil {
+				e2e.Logf("Error detected while accessing the LoadBalancer service on try %d: %q", i, err)
+			} else {
+				results[podName]++
+			}
+		}
+		e2e.Logf("Pods accessed after %d UDP requests:\n%v\n", connNumber, results)
+		pods, err := oc.KubeClient().CoreV1().Pods(oc.Namespace()).List(context.Background(), metav1.ListOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		var successConnCount int
+		for _, pod := range pods.Items {
+			e2e.Logf("Checking if pod '%s' has been accessed...", pod.Name)
+			val, ok := results[pod.Name]
+			o.Expect(ok).To(o.BeTrue(), "Pod '%s' not accessed", pod.Name)
+			e2e.Logf("Pod '%s' has been successfully accessed", pod.Name)
+			successConnCount += val
+		}
```

Comment:
Why do we want to make sure all pods were accessed? I don't see value, IMO if we're super unlucky this may make this test flaky.

Reply:
I've removed it as we don't want it to be flaky due to the balancing between pods, but I'm keeping the check of number of successful connections.